### PR TITLE
Feature/add unit tests

### PR DESCRIPTION
**Pull request by SolovyovDmitry**
Add unit tests for LeaveTime, WorkTime, WorkTimeMonthLimit repositories

### CODE REVIEW

File: src/TimeService.Data/WorkTimeMonthLimitRepository.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public async Task<bool> EditAsync(Guid workTimeMonthLimitId, JsonPatchDocument<D
     {
       DbWorkTimeMonthLimit dbWorkTimeMonthLimit = _provider.WorkTimeMonthLimits.FirstOrDefault(ml => ml.Id == workTimeMonthLimitId);
 
-      if (dbWorkTimeMonthLimit == null)
+      if (dbWorkTimeMonthLimit == null || request == null)
```
Comment (VLiamin) [inline src/TimeService.Data/WorkTimeMonthLimitRepository.cs:57]: May be make is null?

File: src/TimeService.Data/WorkTimeMonthLimitRepository.cs [hunk 2]
```diff
@@ -119,5 +114,10 @@ public async Task<List<DbWorkTimeMonthLimit>> GetAsync(int startYear, int startM
         .Where(ml => ml.Year * 12 + ml.Month >= startCountMonths && ml.Year * 12 + ml.Month <= endCountMonths)
         .ToListAsync();
     }
+
+    public async Task<DbWorkTimeMonthLimit> GetLastAsync()
+    {
+      return await _provider.WorkTimeMonthLimits.OrderByDescending(l => l.Year).ThenByDescending(l => l.Month).FirstOrDefaultAsync();
```
Comment (MariyaSin) [inline src/TimeService.Data/WorkTimeMonthLimitRepository.cs:120]: could you return Task and do not use async/await words?

**Review by MariyaSin — APPROVED** [review]

File: test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+      Assert.IsFalse(await _repository.EditAsync(_secondWorkTimeMonthLimit.Id, null));
+    }
+
+    [Test]
+    public async Task EditReturnFalseByRequestAsync()
+    {
+      await _dbContext.AddAsync(_firstWorkTimeMonthLimit);
+      await _dbContext.SaveChangesAsync();
+
+      Assert.IsFalse(await _repository.EditAsync(_firstWorkTimeMonthLimit.Id, null));
```
Comment (VLiamin) [inline test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs:197]: Why you don not use SerializerAssert? It is our class in UnitTestKernel.

File: test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+      WorkTimeMonthLimitRepository mockedRepository = _mocker.CreateInstance<WorkTimeMonthLimitRepository>();
+
+      string value = "Edited Holidays";
+      JsonPatchDocument<DbWorkTimeMonthLimit> request = new()
+      {
+        Operations = { new() { op = "replace", path = "/Holidays", value = value } }
+      };
+
+      Assert.IsTrue(await mockedRepository.EditAsync(_editableWorkTimeMonthLimit.Id, request));
```
Comment (VLiamin) [inline test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs:178]: Why should not mock class, which we tested.

File: test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs (hunk 3 above)
Comment (SolovyovDmitry) [reply test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs:197]: I noticed it doesn't work correctly and return true when it's actually not

File: test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs (hunk 4 above)
Comment (SolovyovDmitry) [reply test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs:178]: The problem is that I need to mock httpcontextaccessor

**Review by VLiamin — APPROVED** [review]

**Pull request merged**